url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/KitProducerConsumerMonitor.java

```diff
@@ -0,0 +1,75 @@
+package agent;
+
+import java.util.Vector;
+
+import agent.data.Kit;
+
+
+/**
+ * Monitor that protects the shared Kits. Unused in v0.
+ * @author Daniel Paje
+ */
+public class KitProducerConsumerMonitor extends Object {
+	// Since there can only be 3 kits on the stand at any given time.
+	private final int N = 3;
+	private int count = 0;
+	private final Vector<Kit> theData;
+
+	synchronized public void insert(Kit data) {
+		while (count == 4) {
+			try {
+				System.out.println("\tFull, waiting");
+				wait(5000); // Full, wait to add
+			} catch (InterruptedException ex) {
+			}
+			;
+		}
+
+		insert_item(data);
+		count++;
+		if (count == 1) {
+			System.out.println("\tNot Empty, notify");
+			notify(); // Not empty, notify a
+						// waiting consumer
+		}
+	}
+
+	synchronized public Kit remove() {
+		Kit data;
+		while (count == 0) {
+			try {
+				System.out.println("\tEmpty, waiting");
+				wait(5000); // Empty, wait to consume
+			} catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)